### ISSUE TITLE
[oc 9] Reshare not allowed on a fed shared data

### ISSUE THREAD

**Issue by rperezb** [owncloud/core#22500]
### Enviroment
2 servers with the version: ownCloud 9.0 beta 1 (daily) Build:2016-02-18T04:41:04+00:00 4422f9c9ef755ef44385a2ed9e0c42f7d1c459d5

### Steps to reproduce
1. Server1 being logged in as admin, select to share with admin@server2 a folder with "can edit enable"
2. access to admin @ server 2
3. You get a notification about the fed sharing accept it
4. Click on the share icon on the fed sharing folder
5. Share the folder internally, with user2

### Expected behaviour
The folder is shared

### Actual behaviour
There is an error: 

![screen shot 2016-02-18 at 12 21 43](https://cloud.githubusercontent.com/assets/1846927/13141948/3d302fac-d63a-11e5-8993-b3ee9bfdeba7.png)
On the web browser:

```POST http://docker.oc.solidgear.es:55181/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json 403 (Forbidden)```

And the share selector is like:

![screen shot 2016-02-18 at 12 22 49](https://cloud.githubusercontent.com/assets/1846927/13141972/61cf246c-d63a-11e5-9df5-97e9b4b1b8cb.png)

If you select again "test" user, there is an error : path already shared with user

If you refresh: 

![screen shot 2016-02-18 at 12 23 55](https://cloud.githubusercontent.com/assets/1846927/13141993/8283ab74-d63a-11e5-801c-baa7f74de8ed.png)

### Server configuration
ownCloud 9.0 beta 1 (daily) Build:2016-02-18T04:41:04+00:00 4422f9c9ef755ef44385a2ed9e0c42f7d1c459d5
### Client configuration
**Browser:** chrome

**Operating system:**

#### ownCloud log (data/owncloud.log)
```

Error	PHP	OC\User\NoUserException: Backends provided no user object for admin@docker.oc.xxx/ at /opt/owncloud/lib/private/files/filesystem.php#387	2016-02-18T11:23:53+00:00
Error	files	Backends provided no user object for admin@docker.oc.xxxx/	2016-02-18T11:23:53+00:00
```

#### Browser log
```
POST http://docker.oc.xxx/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json 403 (Forbidden)
```

**Comment by rperezb** [comment]
Related to: https://github.com/owncloud/core/issues/22122

**Comment by rullzer** [comment]
Ah I got it. This is actually due to the new resharing behaviour. And it not being implemented for federated shares yet... I think this should not be that hard to fix. I'll take care.

**Comment by rullzer** [comment]
PR in #22503

**Comment by rperezb** [comment]
:+1:

**Comment by nasli** [comment]
:+1: Fixed via #22503
